lib/main.rb

```diff
@@ -1,35 +1,50 @@
 require 'json'
 
 class LogParser
-  attr_reader :data
-
-  def initialize file_path = "games.log"
+  def initialize(file_path = "games.log")
     begin
       @file_path = file_path
-      @data = File.readlines(@file_path, chomp: true)
+      @data = File.readlines(file_path, chomp: true)
     rescue
-      print "Invalid file path."
+      puts "Invalid file path."
     end
   end
 
   def first_line
     @data.first
   end
+
+  def gamelog_json
+    @gamelog_obj_ruby = {
+      @file_path => {
+        lines: line_count(),
+        players: [],
+      } 
+    }
+    player_count
+    gamelog_obj_json = @gamelog_obj_ruby.to_json
+  end
+
+  private
   
   def line_count
-    @lineCount = @data.size
-    result
+    @data.size
   end
 
-  def result
-    result_obj = {
-      "#{@file_path}": {
-        lines: @lineCount
-      } 
-    }
-    result_json = result_obj.to_json
+  def player_count
+    File.foreach(@file_path) do |line|
+      if line.include? "killed"
```

Comment:
aqui da pra usar [guard clause](https://campuscode.com.br/conteudos/guard-clause-em-ruby) dai vc elimina esse bloco if
```suggestion
      next unless line.include? "killed"
```

Reply:
Que excelente! Vou aplicar